smarts/core/sensors.py

```diff
@@ -291,6 +293,11 @@ def observe(sim, agent_id, sensor_state, vehicle):
             sim, agent_id, vehicle, sensor_state
         )
 
+        # print(f'observe steps: {sensor_state.steps_completed} agentId: {agent_id}')
+        # print(f'observe done: {done}')
+        if done and sensor_state.steps_completed == 1 and agent_id == "Agent-007":
```

Comment:
OK, one main thing here.

We cannot rely on the agent id to be `Agent-007` since that is a user field. 


It might be useful to bring in the `AgentIndex` here to query for agents to separate them from the social agents. Then again it may still be worth warning for social agents. I have no hard preference. Someone else may be more opinionated on this issue.